pbft/pbft.cpp

```diff
@@ -1610,6 +1712,64 @@ pbft::proposed_config_is_acceptable(std::shared_ptr<pbft_configuration> /* confi
     return true;
 }
 
+bool
+pbft::adopt_config_from_viewchange(const std::map<uuid_t, bzn_envelope>& viewchange_envelopes)
+{
+    std::map<bzn::hash_t, size_t> config_hashes;
```

Comment:
I don't understand why we're deriving a config from the viewchange set. My understanding was that the new primary chooses a configuration for the newview it builds based on its own local state, and then replicas accept or reject it based on their local state.

Reply:
My thinking was that we should only use a configuration that is acceptable to enough peers, and that we have marked as acceptable ourselves (i.e. enabled).  The alternative would be to simply use our latest enabled configuration regardless. If that's your preference I can change it to do that instead.

Reply:
I'm just concerned about the case where the set of view_change messages does not agree on a configuration. In that case, the view change will fail, and the next time we try, presumably it would be with the same set of configurations? The original logic was designed (through of course it could have issues) such that the new primary would always choose a configuration that would be acceptable to at least 2f+1 replicas- and if we're doing it a different way we need to make sure that's maintained.

Reply:
Changed to always send latest prepared (active) configuration in new_view message